### PR TITLE
SCRUM-1: Create usermanager and provide user storage and retrieval functionality

### PR DESCRIPTION
**Pull request by ovanov**
This change provides commonly used functionality:
- user storage and retrieval
- connection to DynamoDB
- Dockerization
- AWS Lambda functionality

### CODE REVIEW

**Comment by krwro** [comment]
You can move the docker-compose-localstack.yml into the user-manager project folder. In the yml change the context in the user-manager to context: . like this:
  user-manager:
    build:
      context: .
      dockerfile: Dockerfile
      
That way, you can easily run the docker-compose without changing folders in the project terminal.

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public User createUser(User newUser) {
+        if (newUser.getId() == null) {
+            newUser.setId(UUID.randomUUID());
+        }
+
+        Optional<UserInfo> newUserInfo = findUserById(newUser.getId().toString());
+        if (newUserInfo.isEmpty()) {
+            UserInfo createdUserInfo = saveUserToDB(newUser);
+            return mapUserInfoToUser(createdUserInfo);
+        }
```
Comment (krwro) [inline src/main/java/ch/nexusnet/usermanager/service/UserService.java:33]: Should the method silently return the user when that user already exists? I think raising an error telling a user already exists would be more appropriate

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return mapUserInfoToUser(newUserInfo.get());
+    }
+
+    public User getUserByUserId(String userId) throws UserNotFoundException {
+        Optional<UserInfo> userInfo = findUserById(userId);
+        if (userInfo.isPresent()) {
+            return mapUserInfoToUser(userInfo.get());
+        }
+        throw new UserNotFoundException("UserNotFound");
```
Comment (krwro) [inline src/main/java/ch/nexusnet/usermanager/service/UserService.java]: It would be helpful to add the id of the not found user in the Exception message

**Review by krwro — COMMENTED** [review]
Looks good.

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java (hunk 1 above)
Comment (ovanov) [reply src/main/java/ch/nexusnet/usermanager/service/UserService.java:33]: agreed

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java (hunk 2 above)
Comment (ovanov) [reply src/main/java/ch/nexusnet/usermanager/service/UserService.java]: agreed

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+@Service
+@AllArgsConstructor
+@Slf4j
+public class UserService {
+
+    private final UserInfoRepository userInfoRepository;
+
+    public User createUser(User newUser) throws UserAlreadyExistsException {
+        if (newUser.getId() == null) {
+            newUser.setId(UUID.randomUUID());
```
Comment (r-carissimi) [inline src/main/java/ch/nexusnet/usermanager/service/UserService.java:26]: Wouldn't it be better to assign the user's UUID in the constructor?

**Review by r-carissimi — COMMENTED** [review]
Looks good to me

File: src/main/java/ch/nexusnet/usermanager/service/UserService.java (hunk 3 above)
Comment (ovanov) [reply src/main/java/ch/nexusnet/usermanager/service/UserService.java:26]: You can, but when a user is 'created' in the front-end we need to assign a uuid in the back-end. If the front-end can provide the id via amplify, then we can refactor this

Comment (r-carissimi) [reply src/main/java/ch/nexusnet/usermanager/service/UserService.java:26]: Didn't thought about it
Your approach is better than

**Review by krwro — APPROVED** [review]
Looks good!

**Pull request merged**